Default Snowflake JDBC driver throws exception immediately when running queries in 23.3.2

### Description

When running user queries, the execution fails because of an exception in the Snowflake JDBC driver.
This seems to be because of some incompatibility between the predefined Snowflake driver version shipped with DBeaver and DBeaver itself as of 23.3.2.
```
org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [200001] [XX000]: JDBC driver internal error: Fail to retrieve row count for first arrow chunk: null.
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:133)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:600)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$2(SQLQueryJob.java:503)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:190)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:510)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:962)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:4130)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:123)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:190)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:121)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:5148)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: net.snowflake.client.jdbc.SnowflakeSQLLoggedException: JDBC driver internal error: Fail to retrieve row count for first arrow chunk: null.
	at net.snowflake.client.jdbc.SnowflakeResultSetSerializableV1.setFirstChunkRowCountForArrow(SnowflakeResultSetSerializableV1.java:1011)
	at net.snowflake.client.jdbc.SnowflakeResultSetSerializableV1.create(SnowflakeResultSetSerializableV1.java:504)
	at net.snowflake.client.core.SFResultSetFactory.getResultSet(SFResultSetFactory.java:28)
	at net.snowflake.client.core.SFStatement.executeQueryInternal(SFStatement.java:251)
	at net.snowflake.client.core.SFStatement.executeQuery(SFStatement.java:167)
	at net.snowflake.client.core.SFStatement.execute(SFStatement.java:749)
	at net.snowflake.client.jdbc.SnowflakeStatementV1.executeInternal(SnowflakeStatementV1.java:292)
	at net.snowflake.client.jdbc.SnowflakeStatementV1.execute(SnowflakeStatementV1.java:342)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:330)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
	... 12 more
	```

### DBeaver Version

Community Edition 23.3.2

### Operating System

Windows 10

### Database and driver

Snowflake with driver version 3.13.30 as is predefined in DBeaver 23.3.2

Note: Editing the Snowflake settings in Driver Manager in DBeaver and replacing the Snowflake JDBC driver with version 3.14.4 (current as of right now) seems to address the problem

### Steps to reproduce

1. Have DBeaver set up with a Snowflake connection using the predefined driver for Snowflake as-is
2. Run any user query (as opposed to just expanding the tree etc), ie open an editor and run eg `SELECT * FROM mytable`, or even choose "View Data" on a table in the tree.
  


### Additional context

Snowflake JDBC driver version 3.14.4 does appear to work correctly in 23.3.2, it is just not the "prepackaged" version that DBeaver ships

--- a/plugins/org.jkiss.dbeaver.ext.snowflake.core/plugin.xml
+++ b/plugins/org.jkiss.dbeaver.ext.snowflake.core/plugin.xml
@@ -34,7 +34,7 @@
                         categories="sql,analytic"
                         singleConnection="true">
                     <replace provider="generic" driver="snowflake_generic"/>
-                    <file type="jar" path="maven:/net.snowflake:snowflake-jdbc:RELEASE[3.13.30]" bundle="!drivers.snowflake"/>
+                    <file type="jar" path="maven:/net.snowflake:snowflake-jdbc:RELEASE[3.14.4]" bundle="!drivers.snowflake"/>
                     <file type="license" path="https://www.apache.org/licenses/LICENSE-2.0.txt" bundle="!drivers.snowflake"/>
                     <file type="jar" path="drivers/snowflake" bundle="drivers.snowflake"/>
                     <file type="license" path="drivers/apache_lic/LICENSE.txt" bundle="drivers.snowflake"/>

